app/src/main/java/org/mozilla/focus/widget/Cursor.java

```diff
@@ -71,28 +77,44 @@ public float getSpeed() {
 
     public void moveCursor(Direction direction) {
         activeDirections.add(direction);
-        
+
+        // If the cursor isn't moving start the move loop
         if (!moving) {
+
+            // If the cursor is hidden, show it.
+            // If it's currently visible cancel any pending animations
```

Comment:
But what if we're currently animating so we're not at VIEW_MAX_ALPHA? We need to both cancel the animation and setAlpha.

I feel like it'd be easier to just do both unconditionally (within `!moving`).

Reply:
To be explicit, I'm not too concerned about the performance of doing these operations if they're unnecessary because we only do them once. :)